tools/plugins/com.liferay.ide.project.core/src/com/liferay/ide/project/core/model/ParentSDKProjectImportOpMethods.java

```diff
@@ -0,0 +1,150 @@
+/*******************************************************************************
+ * Copyright (c) 2000-present Liferay, Inc. All rights reserved.
+ *
+ * This library is free software; you can redistribute it and/or modify it under
+ * the terms of the GNU Lesser General Public License as published by the Free
+ * Software Foundation; either version 2.1 of the License, or (at your option)
+ * any later version.
+ *
+ * This library is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+ * FOR A PARTICULAR PURPOSE. See the GNU Lesser General Public License for more
+ * details.
+ *
+ *******************************************************************************/
+
+package com.liferay.ide.project.core.model;
+
+import com.liferay.ide.project.core.ProjectCore;
+import com.liferay.ide.sdk.core.SDK;
+import com.liferay.ide.sdk.core.SDKUtil;
+
+import java.util.Map;
+
+import org.eclipse.core.resources.WorkspaceJob;
+import org.eclipse.core.runtime.CoreException;
+import org.eclipse.core.runtime.IPath;
+import org.eclipse.core.runtime.IProgressMonitor;
+import org.eclipse.core.runtime.IStatus;
+import org.eclipse.core.runtime.MultiStatus;
+import org.eclipse.core.runtime.jobs.Job;
+import org.eclipse.sapphire.modeling.Path;
+import org.eclipse.sapphire.modeling.ProgressMonitor;
+import org.eclipse.sapphire.modeling.Status;
+import org.eclipse.sapphire.platform.PathBridge;
+import org.eclipse.sapphire.platform.ProgressMonitorBridge;
+import org.eclipse.sapphire.platform.StatusBridge;
+
+/**
+ * @author Simon Jiang
+ */
+public class ParentSDKProjectImportOpMethods
+{
+
+    public static final Status execute( final ParentSDKProjectImportOp op, final ProgressMonitor pm )
+    {
+        final IProgressMonitor monitor = ProgressMonitorBridge.create( pm );
+
+        monitor.beginTask( "Importing Liferay plugin projects...", 100 );
+
+        Status retval = Status.createOkStatus();
+
+        final Path sdkLocation = op.getSdkLocation().content();
+
+        if( sdkLocation == null || sdkLocation.isEmpty() )
+        {
+            return Status.createErrorStatus( "SDK folder cannot be empty" );
+        }
+
+        final Job job = new WorkspaceJob( "Importing Liferay SDK...")
+        {
+
+            @Override
+            public IStatus runInWorkspace( IProgressMonitor monitor ) throws CoreException
+            {
+                IStatus retval = StatusBridge.create( Status.createOkStatus() );
+
+                final SDK sdk = SDKUtil.createSDKFromLocation( PathBridge.create( sdkLocation ) );
+
+                SDKUtil.openAsProject( sdk );
+
+                return retval;
+            }
+        };
+
+        job.schedule();
+
+        return retval;
+    }
+    
+    public static IStatus validateBuildProperties( final SDK sdk )
```

Comment:
We have other methods that do this same thing nearly exactly teh same, so we should try to re-use those other validation methods instead of duplicating that functionality here.